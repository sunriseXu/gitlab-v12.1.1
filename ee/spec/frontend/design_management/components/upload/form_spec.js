import { shallowMount } from '@vue/test-utils';
import UploadForm from 'ee/design_management/components/upload/form.vue';

describe('Design management upload form component', () => {
  let vm;

  function createComponent(isSaving = false, canUploadDesign = true) {
    vm = shallowMount(UploadForm, {
      propsData: {
        isSaving,
        canUploadDesign,
      },
    });
  }

  afterEach(() => {
    vm.destroy();
  });

  it('renders upload design button', () => {
    createComponent();

    expect(vm.element).toMatchSnapshot();
  });

  it('hides button if cant upload', () => {
    createComponent(false, false);

    expect(vm.element).toMatchSnapshot();
  });

  describe('onFileUploadChange', () => {
    it('emits upload event', () => {
      createComponent();

      vm.vm.onFileUploadChange('test');

      expect(vm.emitted().upload[0]).toEqual(['test']);
    });
  });
});
