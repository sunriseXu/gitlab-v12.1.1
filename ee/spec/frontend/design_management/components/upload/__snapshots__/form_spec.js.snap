// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management upload form component hides button if cant upload 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
>
  <div>
    <!---->
  </div>
</header>
`;

exports[`Design management upload form component renders upload design button 1`] = `
<header
  class="row-content-block border-top-0 p-2 d-flex"
>
  <div>
    <uploadbutton-stub
      isinverted="true"
    />
  </div>
</header>
`;
